backend/rag_service.py

```
@@ -79,19 +83,21 @@ def retrieve(self, query: str, threshold: float = 0.05) -> Optional[str]:
 
-            if not policy_tokens:
+            # Performance: Use isdisjoint for fast early-exit when there is no overlap
+            if query_tokens.isdisjoint(policy_tokens):
                 continue
 
-            # Jaccard Similarity
-            intersection = query_tokens.intersection(policy_tokens)
-            # Use pre-calculated set for union if possible?
-            # Union depends on query_tokens, so must be calculated.
-            union = query_tokens.union(policy_tokens)
+            # Jaccard Similarity: |A ∩ B| / |A ∪ B|
```

Ruff (0.15.12)

[warning] Comment contains ambiguous `∪` (UNION). Did you mean `U` (LATIN CAPITAL LETTER U)?

(RUF003)